cpp/src/io/parquet/page_enc.cu

```diff
@@ -1791,10 +1828,24 @@ CUDF_KERNEL void __launch_bounds__(block_size, 8)
         } break;
 
         case DOUBLE: {
-          auto v = s->col.leaf_column->element<double>(val_idx);
-          memcpy(dst + pos, &v, 8);
+          if (is_split_stream) {
+            auto const v          = s->col.leaf_column->element<int64_t>(val_idx);
+            auto const stride     = s->page.num_valid;
+            dst[pos + 0 * stride] = v;
+            dst[pos + 1 * stride] = v >> 8;
+            dst[pos + 2 * stride] = v >> 16;
+            dst[pos + 3 * stride] = v >> 24;
+            dst[pos + 4 * stride] = v >> 32;
+            dst[pos + 5 * stride] = v >> 40;
+            dst[pos + 6 * stride] = v >> 48;
+            dst[pos + 7 * stride] = v >> 56;
+          } else {
+            auto v = s->col.leaf_column->element<double>(val_idx);
+            memcpy(dst + pos, &v, 8);
```

Comment:
wait, is this the same as the numerous `dst[pos + x] = v >> x*8;` ?

Reply:
Yes. I've consolidated this with INT64.